exercises/circular-buffer/canonical-data.json

```diff
@@ -7,10 +7,10 @@
     "",
     "Tests will contain a number of operations. The operation will be specified in the `operation` key.",
     "Based on the operation, other keys may be present.",
-    "read: Reading from the buffer should produce the item located at `expected`.",
-    "write: Writing the item located at `item` should succeed.",
-    "readShouldFail: Reading from the buffer should fail.",
-    "writeShouldFail: Writing the item located at `item` should fail.",
+    "read: Reading from the buffer should succeed if and only if `should_succeed` is true.",
+    "  If it should succeed, it should produce the item at `expected`. ",
+    "  If it should fail, `expected` will not be present. ",
+    "write: Writing the item located at `item` should succeed if and only if `should_succeed` is true.",
```

Comment:
Do we need to say anything like 'failure may be indicated by raising an exception or similar language dependent runtime error'